@Huang-Wei 
I run perf test like this
```bash
make test-integration WHAT=./test/integration/scheduler_perf KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-alsologtostderr=false -logtostderr=false -run=^$$ -benchtime=1ns -bench=BenchmarkPerfScheduling/TopologySpreading/500Nodes  -data-items-dir ~/logs/$(git rev-parse --abbrev-ref HEAD)/1000"
```
encounter an fail result like this:
```bash
E0406 15:36:15.876368   15557 scheduler.go:342] "Error updating pod" err="Patch \"http://127.0.0.1:38687/api/v1/namespaces/namespace-1/pods/pod-cgcpl/status\": dial tcp 127.0.0.1:38687: connect: connection refused" pod="namespace-1/pod-cgcpl"
--- FAIL: BenchmarkPerfScheduling/PreferredTopologySpreading/500Nodes
    scheduler_perf_test.go:466: op 1: Error creating pod: Failed to create object with non-retriable error: the server has received too many requests and has asked us to try again later (post pods)
--- FAIL: BenchmarkPerfScheduling/PreferredTopologySpreading
--- FAIL: BenchmarkPerfScheduling
FAIL
exit status 1
FAIL    k8s.io/kubernetes/test/integration/scheduler_perf       163.076s
FAIL
make[1]: *** [Makefile:184：test] 错误 1
!!! [0406 15:36:23] Call tree:
!!! [0406 15:36:23]  1: hack/make-rules/test-integration.sh:97 runTests(...)
+++ [0406 15:36:23] Cleaning up etcd
+++ [0406 15:36:23] Integration test cleanup complete
make: *** [Makefile:203：test-integration] 错误 1

```
any indea on this?

--- a/pkg/scheduler/framework/plugins/podtopologyspread/plugin.go
+++ b/pkg/scheduler/framework/plugins/podtopologyspread/plugin.go
@@ -65,6 +65,7 @@ var _ framework.PreFilterPlugin = &PodTopologySpread{}
 var _ framework.FilterPlugin = &PodTopologySpread{}
 var _ framework.PreScorePlugin = &PodTopologySpread{}
 var _ framework.ScorePlugin = &PodTopologySpread{}
+var _ framework.EnqueueExtensions = &PodTopologySpread{}
 
 const (
 	// Name is the name of the plugin used in the plugin registry and configurations.
@@ -118,4 +119,22 @@ func (pl *PodTopologySpread) setListers(factory informers.SharedInformerFactory)
 	pl.replicationCtrls = factory.Core().V1().ReplicationControllers().Lister()
 	pl.replicaSets = factory.Apps().V1().ReplicaSets().Lister()
 	pl.statefulSets = factory.Apps().V1().StatefulSets().Lister()
+}
+
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		// All ActionType includes the following events:
+		// - Add. An unschedulable Pod may fail due to violating topology spread constraints,
+		// adding an assigned Pod may make it schedulable.
+		// - Update. Updating on an existing Pod's labels (e.g., removal) may make
+		// an unschedulable Pod schedulable.
+		// - Delete. An unschedulable Pod may fail due to violating an existing Pod's topology spread constraints,
+		// deleting an existing Pod may make it schedulable.
+		{Resource: framework.Pod, ActionType: framework.All},
+		// Node add|delete|updateLabel maybe lead an topology key changed,
+		// and make these pod in scheduling schedulable or unschedulable.
+		{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.UpdateNodeLabel},
+	}
 }

